packages/expo-firebase-analytics/android/src/main/java/expo/modules/firebase/analytics/FirebaseAnalyticsModule.java

```diff
@@ -0,0 +1,176 @@
+  private static final String NAME = "ExpoFirebaseAnalytics";
+
+  private Activity mActivity;
+  private ModuleRegistry mModuleRegistry;
+
+  public FirebaseAnalyticsModule(Context context) {
+    super(context);
+  }
+
+  @Override
+  public String getName() {
+    return NAME;
+  }
+
+  @Override
+  public void onCreate(ModuleRegistry moduleRegistry) {
+    ActivityProvider mActivityProvider = moduleRegistry.getModule(ActivityProvider.class);
+    mActivity = mActivityProvider.getCurrentActivity();
+    mModuleRegistry = moduleRegistry;
+
+  }
+
+  private FirebaseAnalytics getFirebaseAnalyticsOrReject(final Promise promise) {
+    FirebaseCoreInterface firebaseCore = mModuleRegistry.getModule(FirebaseCoreInterface.class);
+    if (firebaseCore == null) {
+      promise.reject("ERR_FIREBASE_ANALYTICS",
+              "FirebaseCore could not be found. Ensure that your app has correctly linked 'expo-firebase-core' and your project has react-native-unimodules installed.");
+      return null;
+    }
+    FirebaseApp defaultApp = firebaseCore.getDefaultApp();
+    if (defaultApp == null) {
+      promise.reject("ERR_FIREBASE_ANALYTICS",
+              "Firebase app is not initialized. Ensure your app has a valid google-services.json bundled.");
+      return null;
+    }
+    FirebaseApp systemApp = null;
+    try {
+      systemApp = FirebaseApp.getInstance();
+    } catch (Exception e) {
+      // nop
+    }
+    if ((systemApp == null) || !defaultApp.getName().equals(systemApp.getName())) {
+      promise.reject("ERR_FIREBASE_ANALYTICS", "Analytics events can only be logged for the default app.");
+      return null;
+    }
+    FirebaseAnalytics analytics = FirebaseAnalytics.getInstance(mActivity.getApplicationContext());
+    if (analytics == null) {
+      promise.reject("ERR_FIREBASE_ANALYTICS", "Failed to obtain Analytics instance");
+      return null;
+    }
+    return analytics;
+  }
+
+  @ExpoMethod
+  public void logEvent(final String name, @Nullable Map<String, Object> params, Promise promise) {
+    try {
+      FirebaseAnalytics analytics = getFirebaseAnalyticsOrReject(promise);
+      if (analytics == null)
+        return;
+      analytics.logEvent(name, new MapArguments(params).toBundle());
+      promise.resolve(null);
+    } catch (Exception e) {
+      promise.reject(e);
+    }
+  }
+
+  @ExpoMethod
+  public void setAnalyticsCollectionEnabled(final Boolean enabled, Promise promise) {
+    try {
+      FirebaseAnalytics analytics = getFirebaseAnalyticsOrReject(promise);
+      if (analytics == null)
+        return;
+      analytics.setAnalyticsCollectionEnabled(enabled);
+      promise.resolve(null);
+    } catch (Exception e) {
+      promise.reject(e);
+    }
+  }
+
+  @ExpoMethod
+  public void setCurrentScreen(final String screenName, final String screenClassOverride, final Promise promise) {
+    // This is the only method that runs on the main thread.
```

Comment:
interesting.. I wonder why this is required, it seems like the Firebase docs don't really explain it

Reply:
True, it just says: "This method must be called from the app's main thread.".